src/framework/audio/iaudioplayer.h

```diff
@@ -27,13 +27,15 @@
 #include "ret.h"
 
 namespace mu::audio {
-class IAudioPlayer : public virtual IPlayer, public virtual IAudioSource
+class IAudioPlayer : public IPlayer
 {
 public:
     ~IAudioPlayer() = default;
 
     virtual void unload() = 0;
     virtual Ret load(const std::shared_ptr<audio::IAudioStream>& stream) = 0;
+
+    virtual std::shared_ptr<IAudioSource> audioSource() = 0;
```

Comment:
Please, add an alias to this pointer:

using IAudioSourcePtr = std::shared_ptr<IAudioSource>

And add const to this method

Reply:
> using IAudioSourcePtr = std::shared_ptr
I'm not a fan of such aliases, but ok, added

> And add const to this method
unfortunately impossible, does not compile